Development/nmos/api_utils.h

```diff
@@ -125,19 +125,22 @@ namespace nmos
 
     // add handler to set appropriate response headers, and error response body if indicated - call this only after adding all others!
     void add_api_finally_handler(web::http::experimental::listener::api_router& api, slog::base_gate& gate);
+    void add_api_finally_handler(web::http::experimental::listener::api_router& api, const web::http::experimental::hsts& hsts, slog::base_gate& gate);
 
     // modify the specified API to handle all requests (including CORS preflight requests via "OPTIONS") and attach it to the specified listener
     void support_api(web::http::experimental::listener::http_listener& listener, web::http::experimental::listener::api_router api, slog::base_gate& gate);
+    void support_api(web::http::experimental::listener::http_listener& listener, web::http::experimental::listener::api_router api, const web::http::experimental::hsts& hsts, slog::base_gate& gate);
 
     // construct an http_listener on the specified address and port, modifying the specified API to handle all requests
     // (including CORS preflight requests via "OPTIONS")
     web::http::experimental::listener::http_listener make_api_listener(bool secure, const utility::string_t& host_address, int port, web::http::experimental::listener::api_router api, web::http::experimental::listener::http_listener_config config, slog::base_gate& gate);
+    web::http::experimental::listener::http_listener make_api_listener(bool secure, const utility::string_t& host_address, int port, web::http::experimental::listener::api_router api, web::http::experimental::listener::http_listener_config config, const web::http::experimental::hsts& hsts, slog::base_gate& gate);
 
     // construct an http_listener on the specified port, modifying the specified API to handle all requests
     // (including CORS preflight requests via "OPTIONS")
     inline web::http::experimental::listener::http_listener make_api_listener(int port, web::http::experimental::listener::api_router api, web::http::experimental::listener::http_listener_config config, slog::base_gate& gate)
     {
-        return make_api_listener(false, web::http::experimental::listener::host_wildcard, port, api, config, gate);
+        return make_api_listener(false, web::http::experimental::listener::host_wildcard, port, api, config, {}, gate);
```

Comment:
Won't using `{}`, i.e. default-constructed `hsts`, change the behaviour - inserting `Strict-Transport-Security: max-age=0` every time rather than omitting it?

Also applies to all the overloads in _nmos/api_utils.cpp_ too...

Reply:
Resolved by #278.